samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -107,12 +108,14 @@ class LambdaResponseConverter(RemoteInvokeRequestResponseMapper):
     """
 
     def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
-        assert isinstance(remote_invoke_input.response, dict), "Invalid response type received, expecting dict"
-        payload_field = remote_invoke_input.response.get("Payload")
-        if payload_field:
-            remote_invoke_input.response["Payload"] = cast(StreamingBody, payload_field).read().decode("utf-8")
+        if not isinstance(remote_invoke_input.response, dict):
+            raise InvalideBotoResponseException("Invalid response type received from Lambda service, expecting dict")
+        else:
+            payload_field = remote_invoke_input.response.get(PAYLOAD)
+            if payload_field:
+                remote_invoke_input.response[PAYLOAD] = cast(StreamingBody, payload_field).read().decode("utf-8")
 
-        return remote_invoke_input
+            return remote_invoke_input
```

Comment:
nit: for readability, no need to have this `else` block here (just to reduce the indentation), something like;
```
        if not isinstance(remote_invoke_input.response, dict):
            raise InvalideBotoResponseException("Invalid response type received from Lambda service, expecting dict")
        
        payload_field = remote_invoke_input.response.get(PAYLOAD)
        if payload_field:
            remote_invoke_input.response[PAYLOAD] = cast(StreamingBody, payload_field).read().decode("utf-8")

        return remote_invoke_input
```